### DISCUSSION TITLE
Error with NGXS After Update to Version 18: "The selector below has thrown an error upon invocation"

### DISCUSSION THREAD

**Discussion by oscarandrespabonestrella** [ngxs/store#2202] [Q&A]
Hello,

I've encountered an issue after updating NGXS to version 18. My application was working fine before the update, but now I'm seeing the following error:

`The selector below has thrown an error upon invocation. Please check for any unsafe property access that may result in null or undefined values.', state => { ... }
`
This error occurs during our tests, specifically when they are triggered by Jenkins. Interestingly, when I run the tests locally, the error appears, but the tests still pass.

Here are some details about the issue:

- NGXS Version: 18.x
- Angular: 18
- Testing Framework: Jasmine
- Environment: Jenkins CI

Steps I’ve taken so far:

- Reviewed the code to ensure no null or undefined values are being accessed in the selectors.
- Ran the tests locally where the error appears but doesn't cause the tests to fail.
- Despite these steps, I haven't been able to resolve the issue. I would greatly appreciate any advice or suggestions on how to fix this.

Thank you in advance for your help!

UPDATE: I have fixed the issue checking the existence of the state before returning in all selectors across, the app. I don't know if exist a better way to manage this, but was the only one that worked for me

**Comment by SebastianPost1996** [comment]
The default for the `suppressErrors` option in the NgxsModule config has been changed from true to false, meaning that errors in Selectors have previously not been shown. Perhaps the errors have already existed in your project before and this change caused them to show up.

You can adjust the Module config to go back to the old behaviour:

```
NgxsModule.forRoot([...], { selectorOptions: { suppressErrors: true } })
```
